aries-site/src/layouts/content/Subsection.js

```diff
@@ -22,22 +29,50 @@ export const Subsection = ({ children, name, ...rest }) => {
       onFocus={() => setOver(true)}
       onBlur={() => setOver(false)}
     >
-      <Box direction="row" justify="between">
-        <Subheading {...rest}>{name}</Subheading>
-        <Anchor
-          a11yTitle={`Jump to section titled ${name}`}
-          href={`#${id}`}
-          icon={<LinkIcon color={over ? 'text-xweak' : 'transparent'} />}
-        />
+      {/* This condition for gap is needed because the link icon has padding
+       * to maintain a big enough touch area for accessibility. This creates
+       * a larger than desired gap between h3's and its first child. This
+       * removes that extra space.
+       */}
+      <Box gap={GAP_SIZE[level]}>
+        <Box direction="row" justify="between" align="center">
+          <Subheading level={level}>{name}</Subheading>
+          {level > 1 && (
+            <Anchor
+              a11yTitle={`Jump to section titled ${name}`}
+              href={`#${id}`}
+              icon={<LinkIcon color={over ? 'text-xweak' : 'transparent'} />}
+            />
+          )}
+        </Box>
+        {React.Children.map(children, (child, index) => {
+          if (index === 0) {
+            return React.cloneElement(child, {
+              level,
+            });
+          }
+          return undefined;
+        })}
       </Box>
-      {children}
+      {React.Children.map(children, (child, index) => {
+        if (index === 0) {
+          return undefined;
+        }
+        return React.cloneElement(child, {
+          level,
+        });
+      })}
```

Comment:
Aren't those two blocks duplicated? can we extract it to a function to avoid duplication?

Reply:
They're not quite duplicated. The purpose of the first block is to isolate the first child. The first child following the header needs to have the right amount of spacing. In the case where it is an h3, we don't want any gap between the heading and following child because the intrinsic height of the anchor link is creating enough space. The first block renders only the first child. The second block renders everything else.

Reply:
Adding these explanations to the code could be helpful.